cdap-ui/app/hydrator/services/create/stores/config-store.js

```diff
@@ -689,16 +701,31 @@ class HydratorPlusPlusConfigStore {
       });
     };
     outputSchema = nodesMap[pluginId].outputSchema;
+
+    // don't propagate schema if outputSchema contains multiple schema
+    if (this.HydratorPlusPlusNodeService.isMultipleSchema(outputSchema)) {
```

Comment:
Let's add a helper function called `shouldPropagateSchema` that will return true or false whether you should be able to propagate schema or not. We should not propagate the schema if the node type is `batchjoiner` as well.